### PR TITLE
[Data] Move and revise Vale documentation

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

To improve table-of-contents structure, this PR moves "How to use Vale" from a standalone page into the "Contributing to the Ray documentation" page. This PR also adds more information, like how to handle word list-related errors.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by stephanie-wang** [comment]
Don't we need to remove the standalone page too?

**Comment by bveeramani** [comment]
> Don't we need to remove the standalone page too?

Ah, yeah, we do 🤦.

File: doc/source/ray-contribute/docs.ipynb [hunk 1]
```diff
@@ -413,11 +413,120 @@
     "And if you want to convert your notebook to a Python file, e.g. to test if your whole script runs without errors,\n",
     "you can use `--to py` instead.\n",
     "\n",
+    "(vale)=\n",
+    "\n",
+    "## How to use Vale\n",
+    "### What is Vale?\n",
+    "\n",
+    "[Vale](https://vale.sh/) checks if your writing adheres to the \n",
+    "[Google developer documentation style guide](https://developers.google.com/style). Currently,\n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:422]:
```suggestion
    "[Google developer documentation style guide](https://developers.google.com/style).\n",
```

File: doc/source/ray-contribute/docs.ipynb [hunk 2]
```diff
@@ -413,11 +413,120 @@
...
+    "it's only enforced on the Ray Data documentation.\n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:423]:
```suggestion
    "It's only enforced on the Ray Data documentation.\n",
```

File: doc/source/ray-contribute/docs.ipynb [hunk 3]
```diff
@@ -413,11 +413,120 @@
...
+    "\n",
+    "Vale catches typos and grammatical errors. It also enforces stylistic rules like \n",
+    "“use contractions” and “use second person.” For the full list of rules, see the \n",
+    "[configuration in the Ray repository](https://github.com/ray-project/ray/tree/master/.vale/styles/Google).\n",
+    "\n",
+    "### How do you run Vale?\n",
+    "\n",
+    "#### How to use the VSCode extension\n",
+    "\n",
+    "1. Install Vale. If you use MacOS, use HomeBrew.\n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:433]:
```suggestion
    "1. Install Vale. If you use macOS, use Homebrew.\n",
```

File: doc/source/ray-contribute/docs.ipynb [hunk 4]
```diff
@@ -413,11 +413,120 @@
...
+    "2. Install the Vale VSCode extension by following these \n",
+    "[installation instructions](https://marketplace.visualstudio.com/items?itemName=ChrisChinchilla.vale-vscode).\n",
+    "\n",
+    "3. VSCode should show warnings in your code editor and in the “Problems” panel.\n",
+    "    \n",
+    "    ![Vale](../images/vale.png)\n",
+    "        \n",
+    "#### How to run Vale on the command-line\n",
+    "\n",
+    "1. Install Vale. If you use MacOS, use HomeBrew.\n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:456]:
```suggestion
    "1. Install Vale. If you use macOS, use Homebrew.\n",
```

**Review by angelinalg — APPROVED** [review]
Thanks so much for doing this!
The doc build is broken. I'd like to review the rendered version, if possible.

File: doc/source/ray-contribute/docs.ipynb (hunk 2 above)
Comment (bveeramani) [reply doc/source/ray-contribute/docs.ipynb:423]: There's a "Currently," on the preceding line

**Review by stephanie-wang — APPROVED** [review]

**Pull request merged**